ddtrace/internal/wrapping.py

```diff
@@ -370,25 +375,50 @@ def wrap(f, wrapper):
     code.argnames = f.__code__.co_varnames[:nargs]
 
     f.__code__ = code.to_code()
-    wf = cast(WrapperFunction, f)
+
+    # DEV: Multiple wrapping is implemented as a singly-linked list via the
+    # __dd_wrapped__ attribute.
+    wf = cast(WrappedFunction, f)
     wf.__dd_wrapped__ = wrapped
 
     return wf
 
 
-def unwrap(wrapper):
-    # type: (WrapperFunction) -> FunctionType
+def unwrap(wf, wrapper):
```

Comment:
this function is pretty hard to read with all the `cast`s and the variable naming

One idea to help with readability might be to cast the variables before doing any operations which we do in some places, maybe do it wherever possible?